lib/duckdb.js

```diff
@@ -218,12 +225,9 @@ Connection.prototype.each = function (sql) {
  * @param {...*} params
  * @yields row chunks
  */
-Connection.prototype.stream = async function* (sql) {
+Connection.prototype.stream = async function (sql) {
```

Comment:
Description of change:

The  old version used AsyncIterator (function was marked with `*`), which will lead to the wrong return type (not `QueryResult`) + unexpected iteration over another iterator.

Thanks

Reply:
What problem does this solve?
This will break the current API, that is doable but unsure if it makes sense here.
Could the GetColumns not be an async function?

Reply:
It started to return `QueryResult` which is declared in typings, when the old one return `AsyncGenerator`.
It's not possible to call `getColumns` on top of `AsyncIterator`.

```sh
❯ node
Welcome to Node.js v16.19.1.
Type ".help" for more information.
> class MyClass { kek() { return true }; };
undefined
> const fn = async function* () { return new MyClass(); }
undefined
> console.log(fn())
Object [AsyncGenerator] {}
undefined
```

Reply:
I think the current version returns a class wrapping an AsyncGenerator.

Something like:
```js
class MyClass {
        someSyncMethod() {                                                                                 
                return 0;
                }
        async someAsyncMethod() {
                await setTimeout(1);
                return 42;
                }
        async *[Symbol.asyncIterator]() { 
                for (var i = 0; i<10; i++)
                        yield i;
        }
};
```
An instance of this class can behave like an async generator (so `var c = new MyClass; for await (var obj of c) ...`, but can also have methods of its own like `c.someSyncMethod()` or `await c.someAsyncMethod()`.

Reply:
The biggest problem that, old variant is an `AsyncGenerator` on top of `QueryResult` (local scoped), which implements `asyncIterator`.

Reply:
<img width="1051" alt="image" src="https://github.com/duckdb/duckdb-node/assets/572096/ed4d1198-354e-4d11-8772-cba3d5f0bfc4">

![image](https://github.com/duckdb/duckdb-node/assets/572096/08dc378c-ef2a-417a-bdd4-09f7c55f3b51)